erts/emulator/drivers/common/efile_drv.c

```diff
@@ -1290,6 +1340,11 @@ static void invoke_read_file(void *data)
     }
     /* Invariant: d->c.read_file.size >= d->c.read_file.offset */
     
+    if (d->c.read_file.size != 0) {
```

Comment:
Indeed! I am committing a fix now (it all will be squashed anyway, so doesn't matter in the end)
C really needs some RAII 😄